src/components/Form/Form.tsx

```diff
@@ -0,0 +1,55 @@
+import classNames from "classnames";
+import PropTypes from "prop-types";
+import React from "react";
+import type { HTMLProps, ReactNode } from "react";
+
+export type BaseForm = {
+  children?: ReactNode;
+  className?: string;
+};
+
+export type OrderedForm = BaseForm & {
+  inline?: false;
+  stacked?: false;
+};
+
+export type InlineForm = BaseForm & {
+  inline: boolean;
+  stacked?: false;
+};
+
+export type StackedForm = BaseForm & {
+  stacked: boolean;
+  inline?: false;
+};
+
+export type Props = (OrderedForm | InlineForm | StackedForm) &
```

Comment:
This seems a bit hard to parse. I wonder if it would be better to update the props if there are actually three mutually exclusive variants, e.g. `appearance: "default" | "inline" | "stacked" and do the same deprecation thing as with notifications.

Reply:
Yeah I reckon you're right. It's the same thing (only worse) with the list component: https://github.com/canonical-web-and-design/react-components/pull/522.

I suspect as you say we should split these up into individual components, but I'm inclined to want to get these landed as TypeScript files first so that we know if we're breaking things when we split them up (and what the different types should be when we do so).

Reply:
That's fair!